src/compilation/mod.rs

```diff
@@ -1,26 +1,31 @@
 //! Schema compilation.
 //! The main idea is to compile the input JSON Schema to a validators tree that will contain
 //! everything needed to perform such validation in runtime.
+pub(crate) mod config;
+pub(crate) mod context;
+
 use crate::{
     error::{CompilationError, ErrorIterator},
     keywords,
-    keywords::Validators,
+    keywords::{BoxedValidator, Validators},
     resolver::Resolver,
     schemas,
 };
+use config::JSONSchemaCompileConfig;
+use context::CompilationContext;
 use serde_json::Value;
 use std::borrow::Cow;
-use url::{ParseError, Url};
+use url::Url;
 
 pub const DEFAULT_ROOT_URL: &str = "json-schema:///";
 
 /// The structure that holds a JSON Schema compiled into a validation tree
 #[derive(Debug)]
 pub struct JSONSchema<'a> {
-    pub(crate) draft: schemas::Draft,
     pub(crate) schema: &'a Value,
-    pub(crate) validators: Validators,
+    pub(crate) validators: Validators<'static>,
```

Comment:
Why is it `'static` ? It doesn't seem to me, that anything inside `Validators` should have a static lifetime